symbulate/results.py

```diff
@@ -286,22 +287,46 @@ class RVResults(Results):
 
     def plot(self, type=None, alpha=None, normalize=True, jitter=False, 
         bins=30, **kwargs):
+        if type is None:
+            pass
+        elif isinstance(type, str):
+            type = (type,)
+        elif not isinstance(type, (tuple, list)):
+            raise Exception("I don't know how to plot a " + str(type))
+            
         
         dim = get_dimension(self)
         if dim == 1:
             counts = self._get_counts()
             if type is None:
                 heights = counts.values()
                 if sum([(i > 1) for i in heights]) > .8 * len(heights):
-                    type = "impulse"
+                    type = ("impulse",)
                 else:
-                    type = "bar"
-            if type == "bar":
-                if alpha is None:
-                    alpha = .5
-                plt.hist(self, normed=normalize, alpha=alpha, bins=bins, **kwargs)
-                plt.ylabel("Density" if normalize else "Count")
-            elif type == "impulse":
+                    type = ("hist",)
+            if alpha is None:
+                alpha = .5
+            if all(x in ("hist",) for x in type):
+                df = pd.DataFrame({'X':self})
+                ax = df['X'].plot(kind='hist', normed=normalize, alpha=alpha, bins=bins)
+                ax.set_ylabel('')
+            elif all(x in ("density",) for x in type):
+                df = pd.DataFrame({'X':self})
+                ax = df['X'].plot(kind='density', alpha=alpha)
+                ax.set_ylabel('')
+            elif all(x in ("rug",) for x in type):
+                fig, ax = plt.subplots()
+                ax.plot(list(self), [0.001]*len(self), '|', color='k')
+            elif all(x in ("hist", "density") for x in type):
+                df = pd.DataFrame({'X':self})
+                ax = df['X'].plot(kind='hist', normed=normalize, alpha=alpha, bins=bins)
+                df['X'].plot(kind='kde', ax=ax)
+                ax.set_ylabel('')
+            elif all(x in ("hist", "rug") for x in type):
```

Comment:
I'm not sure I like your approach here. You want the user to be able to superimpose any combination of types on the same graphic. You shouldn't have to anticipate all possible combinations. For example, what if the user wants "hist", "density", and "rug" on the same graphic? You haven't anticipated that combination, so that won't work.

Instead of having these conditionals be mutually exclusive, you should just check if a particular type is in `type`, and then add that type of plot to the `Axis` of the graphic.